### PR TITLE
Issue 553: Updating Pravega CRD version to v1

### LINKED ISSUE #553: Add apiVersion check to the CRD
### Description
The apiextensions.k8s.io/v1beta1 apiVersion used in the CustomResourceDefinition was removed in k8s v1.19.0

### Importance
The issue is a blocker to install the chart properly on k8s clusters with version v1.19.0

### Location
charts/pravega-operator/templates/pravega.pravega.io_pravegaclusters_crd.yaml

### Suggestions for an improvement
The issue can be fixed by checking the apiVersion with .Capabilities.APIVersions.Has

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anishakj <anisha.kj@dell.com>

### Change log description

Currently CRD version used for pravega CRD is `v1beta1`. Updating CRD version to `v1`

### Purpose of the change

Fixes #553

### What the code does

Updated operator-sdk to `v0.19.4`
Updated Go version to `1.16`
Updated controller-runtime to `v0.6.5`
Updated CRD vesion, validatingwebhook, clusterrole to `v1`
### How to verify it

Verified that able to install pravega cluster with new CRD

### CODE REVIEW

File: pkg/controller/pravegacluster/upgrade.go [hunk 1]
```diff
@@ -587,6 +601,9 @@ func (r *ReconcilePravegaCluster) scaleSegmentStoreSTS(p *pravegav1beta1.Pravega
 		return fmt.Errorf("updating statefulset (%s) failed due to %v", newsts.Name, err)
 	}
 	*oldsts.Spec.Replicas = *oldsts.Spec.Replicas - 2
+	currentSts := &appsv1.StatefulSet{}
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: oldsts.Name, Namespace: p.Namespace}, currentSts)
```
Comment (jkhalack) [inline pkg/controller/pravegacluster/upgrade.go:606]: Is there a reason for the last two lines of code?
The error is not processed (`err` variable is reassigned on the next line), `currentSts` does not seem to be used anywhere.

Comment (anishakj) [reply pkg/controller/pravegacluster/upgrade.go:606]: Will remove it , added for debugging

Comment (anishakj) [reply pkg/controller/pravegacluster/upgrade.go]: Thanks for finding it, removed it now

**Review by jkhalack — APPROVED** [review]

**Review by nishant-yt — APPROVED** [review]
LGTM

**Pull request merged**